zipline/algorithm.py

```diff
@@ -2416,7 +2421,7 @@ def set_long_only(self, on_error='fail'):
         name=string_types,
         chunks=(int, Iterable, type(None)),
     )
-    def attach_pipeline(self, pipeline, name, chunks=None):
+    def attach_pipeline(self, pipeline, name, chunks=None, is_eager=True):
```

Comment:
I would call this parameter `eager` rather than `is_eager`. `is_eager` suggests that the parameter is a property of the pipeline, whereas `eager` (at least to me) more clearly connotes that this flag describes how we want the pipeline to be computed in the future.